cmd/proxy/sctp_proxy.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"io"
+	"log"
+	"net"
+	"sync"
+
+	"github.com/ishidawataru/sctp"
+)
+
+// SCTPProxy is a proxy for SCTP connections. It implements the Proxy interface to
+// handle SCTP traffic forwarding between the frontend and backend addresses.
+type SCTPProxy struct {
+	listener     *sctp.SCTPListener
+	frontendAddr *sctp.SCTPAddr
+	backendAddr  *sctp.SCTPAddr
+}
+
+// NewSCTPProxy creates a new SCTPProxy.
+func NewSCTPProxy(frontendAddr, backendAddr *sctp.SCTPAddr) (*SCTPProxy, error) {
+	listener, err := sctp.ListenSCTP("sctp", frontendAddr)
+	if err != nil {
+		return nil, err
+	}
+	// If the port in frontendAddr was 0 then ListenSCTP will have a picked
+	// a port to listen on, hence the call to Addr to get that actual port:
```

Comment:
do you mean that if the port is not specified is ephemeral?

Reply:
yes. the comment is taken from `tcp_proxy.go`.